### PR TITLE
Made Api endpoint for ItemRecommendations

### LINKED ISSUE #26: Prepare back-end calls to get results

### PR DESCRIPTION
**Pull request by kryczkal**

### CODE REVIEW

File: frontend/src/api/apiCalls.ts [hunk 1]
```diff
@@ -1,28 +1,42 @@
...
   try {
-    const response = await fetch(API_ENDPOINTS.validateSearchResult, {
-      method: 'GET',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-    });
-    if(!response.ok) {
-      throw new Error('Failed to fetch results');
+    const response = await axios.post<ValidSearchPromptResponse>(API_ENDPOINTS.validateSearchResult, { prompt });
```
Comment (Wflikeit) [inline frontend/src/api/apiCalls.ts:14]: export async function validateSearchPrompt(prompt: string): Promise<boolean> {
  try {
    const response = await axios.post(API_ENDPOINTS.validateSearchResult, { prompt });
    return response.data.status === 'success';
  } catch (error) {
    console.error('Error validating search prompt:', error); // Log the entire error for debugging
  }
  return false;
}

File: frontend/src/api/apiCalls.ts [hunk 2]
```diff
@@ -1,28 +1,42 @@
...
+  recommendations: ItemRecommendationBoxData[];
+}
 
-    return true;
+export async function getRecommendations(prompt : string) : Promise<ItemRecommendationBoxData[]> {
+  try {
+    const response = await axios.post<RecommendationResponse>(API_ENDPOINTS.getRecommendations, { prompt });
+    return response.data.recommendations;
   } catch (error) {
     if (error instanceof Error) {
```
Comment (Wflikeit) [inline frontend/src/api/apiCalls.ts]: make sure that this check is needed

Comment (Jlisowskyy) [reply frontend/src/api/apiCalls.ts]: solved

**Review by Jlisowskyy — APPROVED** [review]
should work well

**Pull request merged**